benchopt/plotting/helpers_compat.py

```diff
@@ -55,3 +56,43 @@ def add_h_line(fig, val, xlim=None, plotly=False):
         line_dash='dot', line_color='black',
         mode='lines', showlegend=False
     ))
+
+
+def make_bars(fig, heights, ticks, width, colors, times, plotly=False):
+    if not plotly:
+        ax = fig.gca()
+        for idx, tick in enumerate(ticks):
+            xi = tick[0]
+            height = heights[idx]
+            edges = colors[idx] if colors[idx] != "w" else "k"
+            ax.bar(x=xi, height=height, width=width,
+                   color=colors[idx], edgecolor=edges)
+            if colors[idx] == "w":
+                ax.text(xi, .5, "Did not converge",
+                        ha="center", va="center", color='k',
+                        rotation=90, transform=ax.transAxes)
+            else:
+                plt.scatter(np.ones_like(times[idx]) * xi, times[idx],
+                            marker='_', color='k', zorder=10)
+    else:
+        colors = [f'rgba{color}' for color in colors]
+        xi, _ = zip(*ticks)
+        no_cv = [True if col ==
+                 "rgba(0.8627, 0.8627, 0.8627)" else False for col in colors]
```

Comment:
Not a big fan of checking for "did not converged" based on the color.
it seems error prone. isn't it possible to use missing value for heights?

Reply:
Yes it's possible. I changed it with missing values for times (as heights is set to the max if it didn't converge).